misc/plot-bamstats

```diff
@@ -1204,6 +1207,119 @@ sub plot_indel_cycles
     plot($$args{gp});
 }
 
+sub plot_read_len_hist
+{
+    my ($opts) = @_;
+
+    my @rl = get_values($opts, 'RL');
+    if ( !@rl ) { return; }
+
+    # TODO: Add modified version for paired end reads.
+    my $is_paired = get_value($opts,'SN','reads paired:');
+
+    my $args = get_defaults($opts,"$$opts{prefix}read_length.png");
+    my $fh = $$args{fh};
+
+    # anonymous subroutines for code readibility
+    my $log_bin = sub {
+        my $x = $_[0];
+        return floor((log($x) / log(10)) * 10);
+    };
+    my $inv_log_bin = sub {
+        my $x = $_[0];
+        return floor(10**($x/10));
+    };
+
+    # Open files to store data: this is more elegant than copying four sets of 
+    # data into a single GNUplot file.
+    open(my $rd_fh, '>', "read_len_rawdata.txt");
+    open(my $hd_fh, '>', "read_len_histdata.txt");
+    
+    # Create gnuplot command file options
+    my $min_x = $rl[0][0];
+    my $max_x = $rl[-1][0];
+    # Gnuplot commands for normalized binned data
+    print $fh qq[
+        set y2tic
```

Comment:
We could do with `set ytics nomirror` here too as the right Y axis has two sets of tics combined on it.